gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetTest.java

```diff
@@ -176,22 +176,72 @@ public void testGenerateCopyEntitiesMultiSnapshotWhenDestEmpty() throws IOExcept
+    IcebergDataset icebergDataset = new TrickIcebergDataset(testDbName, testTblName, icebergTable, new Properties(), sourceFs);
+
+    MockFileSystemBuilder destBuilder = new MockFileSystemBuilder(DEST_FS_URI);
+    FileSystem destFs = destBuilder.build();
+
+    CopyConfiguration copyConfiguration = CopyConfiguration.builder(destFs, copyConfigProperties)
+        .preserve(PreserveAttributes.fromMnemonicString("ugp"))
+        .copyContext(new CopyContext())
+        .build();
+    Collection<CopyEntity> copyEntities = icebergDataset.generateCopyEntities(destFs, copyConfiguration);
+    verifyFsOwnershipAndPermissionPreservation(copyEntities);
+  }
+
   private void verifyCopyEntities(Collection<CopyEntity> copyEntities, List<String> expected) {
     List<String> actual = new ArrayList<>();
     for (CopyEntity copyEntity : copyEntities) {
       String json = copyEntity.toString();
-      String filepath = new Gson().fromJson(json, JsonObject.class)
-          .getAsJsonObject("object-data").getAsJsonObject("origin")
-          .getAsJsonObject("object-data").getAsJsonObject("path")
-          .getAsJsonObject("object-data").getAsJsonObject("uri")
-          .getAsJsonPrimitive("object-data").getAsString();
+      String filepath = getFilePathAsStringFromJson(json);
       actual.add(filepath);
     }
     Assert.assertEquals(actual.size(), expected.size(),
         "Set" + actual.toString() + " vs Set" + expected.toString());
     Assert.assertEqualsNoOrder(actual.toArray(), expected.toArray());
   }
 
+  private void verifyFsOwnershipAndPermissionPreservation(Collection<CopyEntity> copyEntities) {
+    List<JsonObject> ancestorsOwnerAndPermissionsList = new ArrayList<>();
+    List<JsonObject> destinationOwnerAndPermissionsList = new ArrayList<>();
+
+    for(CopyEntity copyEntity : copyEntities) {
+      String copyEntityJson = copyEntity.toString();
+
+      JsonArray ancestorsOwnerAndPermissions = new Gson().fromJson(copyEntityJson, JsonObject.class)
+          .getAsJsonObject("object-data").getAsJsonArray("ancestorsOwnerAndPermission");
+      JsonObject rootAncestorOwnerAndPermissions = ancestorsOwnerAndPermissions.get(ancestorsOwnerAndPermissions.size()-1)
+          .getAsJsonObject().getAsJsonObject();
+      ancestorsOwnerAndPermissionsList.add(rootAncestorOwnerAndPermissions);
+      String filePath = getFilePathAsStringFromJson(copyEntityJson);
+      Assert.assertEquals(ancestorsOwnerAndPermissions.size(), new Path(filePath).getParent().depth() -1);
+
+      JsonObject destinationOwnerAndPermissions = new Gson().fromJson(copyEntityJson, JsonObject.class)
+          .getAsJsonObject("object-data").getAsJsonObject("destinationOwnerAndPermission");
+      destinationOwnerAndPermissionsList.add(destinationOwnerAndPermissions);
+    }
+    Assert.assertEqualsNoOrder(ancestorsOwnerAndPermissionsList.toArray(), destinationOwnerAndPermissionsList.toArray());
```

Comment:
both of these were initialized from the `CopyEntity` JSON... so, perhaps--maybe?--that demonstrates some level of internal coherence in the result... nonetheless it doesn't verify that the particular values of either are anything in particular (like what an observer would expect them to be).